62.CAD/fragment_shader.hlsl

```diff
@@ -52,12 +64,29 @@ float4 main(PSInput input) : SV_TARGET
         const float antiAliasingFactor = globals.antiAliasingFactor;
         localAlpha = 1.0f - smoothstep(-antiAliasingFactor, +antiAliasingFactor, distance);
     }
-    /*
-    TODO[Lucas]:
-        Another else case for CurveBox where you simply do what I said in the notes of common.hlsl PSInput
-        and solve two quadratic equations, you could check for it being a "line" for the mid point being nan
-        you will use input.getXXX() to get values needed for this computation
-    */
+    else if (objType == ObjectType::CURVE_BOX) 
+    {
+        float uv = input.getUVMinor();
+
+        float minT = tForMajorCoordinate(input.getMinCurveDetRcp2(), input.getMinCurveBrcp());
+        float minEv = evaluateBezier(input.getCurveMinA().y, input.getCurveMinB().y, input.getCurveMinC().y, minT);
+        
+        float maxT = tForMajorCoordinate(input.getMaxCurveDetRcp2(), input.getMaxCurveBrcp());
+        float maxEv = evaluateBezier(input.getCurveMaxA().y, input.getCurveMaxB().y, input.getCurveMaxC().y, maxT);
```

Comment:
you only need to evaluate bezier at minor axis, 
use `nbl::hlsl::equation::Quadratic::evaluate` and make those input values floats instead of float2